### PR TITLE
save tx hash on order creation 

### PR DESCRIPTION
**Pull request by Meketh**
### What does this Pull Request do?
Saves transaction hash on order creation.

### Are there points in the code the reviewer needs to double check?
There are some unrelated improvements on blockchain/contracts that may be needed on the future.

### Why was this Pull Request needed?
Necesary to send metadata on BatPay.Transfer.

### Does this Pull Request meet the acceptance criteria?
- [x] I have read the [Contribution guidelines](https://github.com/wibsonorg/buyer-sdk/CONTRIBUTING.md).
- [x] I have read the [Code of Conduct](https://github.com/wibsonorg/buyer-sdk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: buyer-api/src/blockchain/contractEventSubscribers.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * @property {number} orderId The DataExchange id of the DataOrder
+
+ * @callback dataOrderUpdapter Updates DataOrder in the store with data from the DataExchange
+ * @param {DataOrderEventValues} eventValues The values emmited by the DataExchange event
+
+ * @function createDataOrderUpdapter Creates a dataOrderUpdapter with the given status
+ * @param {string} status The status to be set by the updater
+ * @returns {dataOrderUpdapter}
+ */
+const createDataOrderUpdapter = status => async ({ owner, orderId: dxId }, { transactionHash }) => {
```
Comment (nicoayala) [inline buyer-api/src/blockchain/contractEventSubscribers.js]:
```suggestion
const createDataOrderUpdater = status => async ({ owner, orderId: dxId }, { transactionHash }) => {
```

File: buyer-api/src/blockchain/contractEventSubscribers.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        status,
+      });
+      apicache.clear('/orders/*');
+    }
+  }
+};
+
+contractEventListener
+  .addContract(DataExchange)
+  .on('DataOrderCreated', createDataOrderUpdapter('created'))
```
Comment (nicoayala) [inline buyer-api/src/blockchain/contractEventSubscribers.js]:
```suggestion
  .on('DataOrderCreated', createDataOrderUpdater('created'))
```

File: buyer-api/src/blockchain/contractEventSubscribers.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  .on('DataOrderClosed', createDataOrderUpdapter('closed'));
```
Comment (nicoayala) [inline buyer-api/src/blockchain/contractEventSubscribers.js]:
```suggestion
  .on('DataOrderClosed', createDataOrderUpdater('closed'));
```

File: buyer-api/src/blockchain/contractEventSubscribers.js (hunk 1 above)
Comment (genuinefafa) [reply buyer-api/src/blockchain/contractEventSubscribers.js]: well done @nicoayala

**Pull request merged**